lang/cs/Org.Apache.REEF.Wake/Time/Time.cs

```diff
@@ -50,33 +55,21 @@ public override bool Equals(object obj)
             {
                 return true;
             }
-            Time other = obj as Time;
+            
+            // Note: This allows for the very strange semantic where
+            // two different subclasses might return true for Equals.
+            var other = obj as Time;
```

Comment:
Let's do this as a separate discussion. The change as is is minimally invasive and fixes a bug, so we should proceed.